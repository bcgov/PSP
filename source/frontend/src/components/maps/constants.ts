import { LatLngBounds } from 'leaflet';

export const DEFAULT_MAP_ZOOM = 6;
/** rough center of bc Itcha Ilgachuz Provincial Park */
export const defaultLatLng = {
  lat: 52.81604319154934,
  lng: -124.67285156250001,
};

// default BC map bounds
export const defaultBounds = new LatLngBounds(
  [60.09114547, -119.49609429],
  [48.78370426, -139.35937554],
);

export const ONE_HUNDRED_METER_PRECISION = 0.0001;
