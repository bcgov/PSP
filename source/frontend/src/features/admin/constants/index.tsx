import { ColumnWithProps } from 'components/Table';
import { IOrganization, IOrganizationRecord } from 'interfaces';
import * as React from 'react';
import { Link } from 'react-router-dom';
import { CellProps } from 'react-table';

export const columnDefinitions: ColumnWithProps<IOrganizationRecord>[] = [
  {
    Header: 'Organization name',
    accessor: 'name',
    align: 'left',
    clickable: true,
  },
  {
    Header: 'Short name',
    accessor: 'code',
    align: 'left',
    clickable: true,
  },
  {
    Header: 'Description',
    accessor: 'description',
    align: 'left',
    clickable: true,
  },
  {
    Header: 'Parent Organization',
    accessor: 'parent',
    align: 'left',
    clickable: true,
    Cell: (props: CellProps<IOrganization>) => {
      return (
        <Link to={`/admin/organization/${props.row.original.parentId}`}>
          {props.row.original.parent}
        </Link>
      );
    },
  },
];
