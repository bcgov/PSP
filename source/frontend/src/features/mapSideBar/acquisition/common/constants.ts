export const cannotEditMessage =
  'This section of the file is in a non-editable state. Contact a System administrator to change it to an editable state';
