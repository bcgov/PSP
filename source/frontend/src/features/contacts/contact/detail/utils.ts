import { ContactInfoField } from 'features/contacts/interfaces';
import { Dictionary } from 'interfaces/Dictionary';
import { IContactAddress, IContactMethod } from 'interfaces/IContact';

import { AddressTypes } from './../../../../constants/addressTypes';

// the order of this array corresponds to the expected display order
const addressSortOrder = [AddressTypes.Mailing, AddressTypes.Residential, AddressTypes.Billing];

export const sortAddresses = (a1: IContactAddress, a2: IContactAddress) => {
  if (a2.addressType.id === a1.addressType.id) {
    return 0;
  }
  const a2Index = addressSortOrder.indexOf(a2.addressType.id as AddressTypes);
  if (a2Index !== -1 && a2Index < addressSortOrder.indexOf(a1.addressType.id as AddressTypes)) {
    return 1;
  }

  return -1;
};

export function getContactInfo(
  validTypes: Dictionary<string>,
  contactMethods?: IContactMethod[],
): ContactInfoField[] {
  if (contactMethods === undefined) {
    return [];
  }
  // Get only the valid types
  let filteredFields = contactMethods.reduce(
    (accumulator: ContactInfoField[], method: IContactMethod) => {
      if (Object.keys(validTypes).includes(method.contactMethodType.id)) {
        accumulator.push({
          info: method.value,
          label: validTypes[method.contactMethodType.id],
        });
      }
      return accumulator;
    },
    [],
  );

  // Sort according to the dictionary order
  return filteredFields.sort((a, b) => {
    return Object.values(validTypes).indexOf(a.label) - Object.values(validTypes).indexOf(b.label);
  });
}

export const fakeAddresses = [
  {
    id: 1,
    rowVersion: 3,
    addressType: {
      id: 'BILLING',
      description: 'Billing address',
      isDisabled: false,
    },
    streetAddress1: 'Billing address',
    municipality: 'Hollywood North',
    province: {
      provinceStateId: 1,
      provinceStateCode: 'BC',
      description: 'British Columbia',
    },
    country: {
      countryId: 1,
      countryCode: 'CA',
      description: 'Canada',
    },
    postal: 'V6A 5G7',
  },
  {
    id: 1,
    rowVersion: 3,
    addressType: {
      id: 'MAILING',
      description: 'Mailing address',
      isDisabled: false,
    },
    streetAddress1: 'Mailing address',
    streetAddress2: 'Living in a van',
    streetAddress3: 'Down by the River',
    municipality: 'Hollywood North',
    province: {
      provinceStateId: 1,
      provinceStateCode: 'BC',
      description: 'British Columbia',
    },
    country: {
      countryId: 1,
      countryCode: 'CA',
      description: 'Canada',
    },
    postal: 'V6Z 5G7',
  },
  {
    id: 5,
    rowVersion: 1,
    addressType: {
      id: 'RESIDNT',
      description: 'Property address',
      isDisabled: false,
    },
    streetAddress1: 'Property address',
    streetAddress2: '',
    streetAddress3: '',
    municipality: 'vic',
    province: {
      provinceStateId: 1,
      provinceStateCode: 'BC',
      description: 'British Columbia',
    },
    country: {
      countryId: 1,
      countryCode: 'CA',
      description: 'Canada',
    },
    postal: 'V9A 0H6',
  },
];
