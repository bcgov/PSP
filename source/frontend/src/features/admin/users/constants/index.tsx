import { ReactComponent as Active } from 'assets/images/active.svg';
import { ReactComponent as Inactive } from 'assets/images/inactive.svg';
import { ColumnWithProps } from 'components/Table';
import { DateTimeCell } from 'components/Table/DateCell';
import { Link } from 'react-router-dom';
import { CellProps } from 'react-table';
import { stringToFragment } from 'utils';

import { RowActions } from '../components/RowActions';
import { FormUser } from '../models';

export const getUserColumns = (refresh: () => void): ColumnWithProps<FormUser>[] => [
  {
    Header: 'Active',
    accessor: 'isDisabled',
    align: 'left',
    clickable: true,
    sortable: true,
    width: 60,
    Cell: (props: CellProps<FormUser>) =>
      props.row.original.isDisabled ? <Inactive /> : <Active />,
  },
  {
    Header: 'IDIR/BCeID',
    accessor: 'businessIdentifierValue',
    align: 'left',
    minWidth: 150,
    Cell: (props: CellProps<FormUser>) => {
      return (
        <Link to={`/admin/user/${props.row.original.id}`}>
          {props.row.original.businessIdentifierValue}
        </Link>
      );
    },
    sortable: true,
  },
  {
    Header: 'First name',
    accessor: 'firstName',
    align: 'left',
    clickable: true,
    sortable: true,
  },
  {
    Header: 'Last name',
    accessor: 'surname',
    align: 'left',
    clickable: true,
    sortable: true,
  },
  {
    Header: 'Email',
    accessor: 'email',
    align: 'left',
    clickable: true,
    sortable: true,
    minWidth: 200,
  },
  {
    Header: 'Position',
    accessor: 'position',
    align: 'left',
    clickable: true,
    sortable: true,
  },

  {
    Header: 'Roles',
    accessor: 'roles',
    align: 'left',
    clickable: true,
    minWidth: 200,
    Cell: (props: CellProps<FormUser>) => {
      const rolesString = props.row?.original?.roles?.map(userRole => userRole?.name);
      return stringToFragment(rolesString?.join(', '));
    },
  },
  {
    Header: 'MoTI region(s)',
    accessor: 'regions',
    align: 'left',
    clickable: true,
    minWidth: 200,
    Cell: (props: CellProps<FormUser>) => {
      const regionsString = props.row?.original?.regions?.map(
        regionUser => regionUser?.description,
      );
      return stringToFragment(regionsString?.join(', '));
    },
  },
  {
    Header: 'Last login',
    accessor: 'lastLogin',
    align: 'left',
    clickable: true,
    minWidth: 100,
    Cell: DateTimeCell,
  },
  {
    Header: ' ',
    Cell: (props: CellProps<FormUser>) => {
      return <RowActions {...props} refresh={refresh} />;
    },
    width: 75,
  },
];
