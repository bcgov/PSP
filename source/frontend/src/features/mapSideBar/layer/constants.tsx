import { Feature } from 'geojson';
import { FaExternalLinkAlt } from 'react-icons/fa';

import { ApiGen_Concepts_Address } from '@/models/api/generated/ApiGen_Concepts_Address';
import { exists, formatApiAddress, getAreaUnit, pidFormatter } from '@/utils';

import { ContentConfig } from './LayerContent';

export const parcelLayerConfig: ContentConfig = {
  PID: { label: 'Parcel PID', display: (data: { [key: string]: any }) => pidFormatter(data?.PID) },
  PIN: { label: 'Parcel PIN', display: (data: { [key: string]: any }) => data.PIN },
  PLAN_NUMBER: {
    label: 'Plan number',
    display: (data: { [key: string]: any }) => data.PLAN_NUMBER,
  },
  LEGAL_DESCRIPTION: {
    label: 'Legal description',
    display: (data: { [key: string]: any }) => data.LEGAL_DESCRIPTION,
  },
  OWNER_TYPE: { label: 'Owner type', display: (data: { [key: string]: any }) => data.OWNER_TYPE },
  MUNICIPALITY: {
    label: 'Municipality',
    display: (data: { [key: string]: any }) => data.MUNICIPALITY,
  },
  REGIONAL_DISTRICT: {
    label: 'Regional district',
    display: (data: { [key: string]: any }) => data.REGIONAL_DISTRICT,
  },
  FEATURE_AREA_SQM: {
    label: 'Area',
    display: (data: { [key: string]: any }) => (
      <>
        {data.FEATURE_AREA_SQM} m<sup>2</sup>
      </>
    ),
  },
  PARCEL_CLASS: {
    label: 'Parcel class',
    display: (data: { [key: string]: any }) => data.PARCEL_CLASS,
  },
  SURVEY_DESIGNATION_1: {
    label: 'Survey designation one',
    display: (data: { [key: string]: any }) => data.SURVEY_DESIGNATION_1,
  },
  SURVEY_DESIGNATION_2: {
    label: 'Survey designation two',
    display: (data: { [key: string]: any }) => data.SURVEY_DESIGNATION_2,
  },
  SURVEY_DESIGNATION_3: {
    label: 'Survey designation three',
    display: (data: { [key: string]: any }) => data.SURVEY_DESIGNATION_3,
  },
};

// Utility function that dynamically creates a ContentConfig for a feature.
// Transforms "THE_PARAM" to "The param"
export const getDynamicFeatureConfig: (feature: Feature) => ContentConfig = (feature: Feature) => {
  // For debug enable this
  /*
  config['Feature Type'] = {
    label: 'Feature Type',
    display: (data: { [key: string]: any }) => feature.id,
  };
  */
  const config: ContentConfig = {};
  Object.keys(feature.properties).forEach(key => {
    config[key] = {
      label: key.charAt(0).toUpperCase() + key.slice(1).toLowerCase().replaceAll('_', ' '),
      display: (data: { [key: string]: any }) => data[key],
    };
  });
  // This is not being mapped in the typed feature data
  if (exists(config['SE_ANNO_CAD_DATA'])) {
    delete config['SE_ANNO_CAD_DATA'];
  }
  return config;
};

export const municipalityLayerConfig: ContentConfig = {
  MUNICIPALITY_NAME: {
    label: 'Municipality Name',
    display: (data: { [key: string]: any }) =>
      `${data.ADMIN_AREA_GROUP_NAME} (${data.CHANGE_REQUESTED_ORG})`,
  },
};

export const pimsLayerConfig: ContentConfig = {
  PROPERTY_ID: {
    label: 'Hyperlink',
    display: (data: { [key: string]: any }) => (
      <a
        href={`/mapview/sidebar/property/${data.PROPERTY_ID}/details`}
        target="_blank"
        rel="noreferrer"
      >
        View PIMS property&nbsp;
        <FaExternalLinkAlt />
      </a>
    ),
  },
  PID: {
    label: 'PID',
    display: (data: { [key: string]: any }) => pidFormatter(data.PID_PADDED),
  },
  PIN: {
    label: 'PIN',
    display: (data: { [key: string]: any }) => data.PIN,
  },
  UNIQUE_ID: {
    label: 'Plan number',
    display: (data: { [key: string]: any }) => data.SURVEY_PLAN_NUMBER,
  },
  IS_RETIRED: {
    label: 'Retired ?',
    display: (data: { [key: string]: any }) => (data.IS_RETIRED ? 'Yes' : 'No'),
  },
  LAND_LEGAL_DESCRIPTION: {
    label: 'PIMS legal description',
    display: (data: { [key: string]: any }) => data.LAND_LEGAL_DESCRIPTION,
  },

  PIMS_ADDRESS: {
    label: 'Pims Address',
    display: (data: { [key: string]: any }) =>
      formatApiAddress({
        streetAddress1: data.STREET_ADDRESS_1,
        streetAddress2: data.STREET_ADDRESS_2,
        streetAddress3: data.STREET_ADDRESS_3,
        postal: data.POSTAL_CODE,
        province: { code: 'BC' },
        municipality: data.MUNICIPALITY_NAME,
      } as ApiGen_Concepts_Address), // force the type to avoid having to list unused fields.
  },
  LAND_AREA: {
    label: 'PIMS land area',
    display: (data: { [key: string]: any }) =>
      data.LAND_AREA + ' ' + getAreaUnit(data.PROPERTY_AREA_UNIT_TYPE_CODE.toLowerCase()),
  },
};

export const highwayLayerConfig: ContentConfig = {
  PROVINCIAL_PUBLIC_HIGHWAY_ID: {
    label: 'Id',
    display: (data: { [key: string]: any }) => data.PROVINCIAL_PUBLIC_HIGHWAY_ID,
  },
  GLOBALID: { label: 'Global Id', display: (data: { [key: string]: any }) => data.GLOBALID },
  UNIQUE_ID: { label: 'Unique Id', display: (data: { [key: string]: any }) => data.UNIQUE_ID },
  HYPERLINK: {
    label: 'Hyperlink to plan documents',
    display: (data: { [key: string]: any }) => (
      <a href={`${data.HYPERLINK}`} target="_blank" rel="noreferrer">
        {decodeURI(data.HYPERLINK)}
      </a>
    ),
  },
  PLAN_ANNOTATION: {
    label: 'Plan annotation',
    display: (data: { [key: string]: any }) => data.PLAN_ANNOTATION,
  },
  MOTI_PLAN: { label: 'MOTI plan', display: (data: { [key: string]: any }) => data.MOTI_PLAN },
  VETTING_STATUS: {
    label: 'Vetting Status',
    display: (data: { [key: string]: any }) => data.VETTING_STATUS,
  },
  SHAPE_TYPE: { label: 'Shape type', display: (data: { [key: string]: any }) => data.SHAPE_TYPE },
  PENDING_CLASSIFICATION: {
    label: 'Pending classification',
    display: (data: { [key: string]: any }) => data.PENDING_CLASSIFICATION,
  },
  MOTI_FILE: { label: 'MOTI file', display: (data: { [key: string]: any }) => data.MOTI_FILE },
  GAZETTE_PUBLISHED_DATE: {
    label: 'Gazette published date',
    display: (data: { [key: string]: any }) => data.GAZETTE_PUBLISHED_DATE,
  },
  GAZETTE_PUBLISHED_ON: {
    label: 'Gazette published on',
    display: (data: { [key: string]: any }) => data.GAZETTE_PUBLISHED_ON,
  },
  PROV_PUBLIC_HIGHWAY_TYPE: {
    label: 'Prov public highway type',
    display: (data: { [key: string]: any }) => data.PROV_PUBLIC_HIGHWAY_TYPE,
  },
  ORDER_IN_COUNCIL: {
    label: 'Order in council',
    display: (data: { [key: string]: any }) => data.ORDER_IN_COUNCIL,
  },
  SHORT_LEGAL_DESCRIPTION: {
    label: 'Short legal description',
    display: (data: { [key: string]: any }) => data.SHORT_LEGAL_DESCRIPTION,
  },
  SURVEY_PLAN: {
    label: 'Survey plan',
    display: (data: { [key: string]: any }) => data.SURVEY_PLAN,
  },
  PARENT_PARCEL: {
    label: 'Parent parcel',
    display: (data: { [key: string]: any }) => data.PARENT_PARCEL,
  },
  PARCELMAPBC_COMMENT: {
    label: 'Parcelmapbc comment',
    display: (data: { [key: string]: any }) => data.PARCELMAPBC_COMMENT,
  },
  CREATE_TIME: {
    label: 'Create time',
    display: (data: { [key: string]: any }) => data.CREATE_TIME,
  },
  UPDATE_TIME: {
    label: 'Update time',
    display: (data: { [key: string]: any }) => data.UPDATE_TIME,
  },
  CREATE_USER: {
    label: 'Create user',
    display: (data: { [key: string]: any }) => data.CREATE_USER,
  },
  UPDATE_USER: {
    label: 'Update user',
    display: (data: { [key: string]: any }) => data.UPDATE_USER,
  },
  LTSA_CREATE_TIME: {
    label: 'Ltsa create time',
    display: (data: { [key: string]: any }) => data.LTSA_CREATE_TIME,
  },
  LTSA_UPDATE_TIME: {
    label: 'Ltsa update time',
    display: (data: { [key: string]: any }) => data.LTSA_UPDATE_TIME,
  },
  LTSA_CREATE_USER: {
    label: 'Ltsa create user',
    display: (data: { [key: string]: any }) => data.LTSA_CREATE_USER,
  },
  LTSA_UPDATE_USER: {
    label: 'Ltsa update user',
    display: (data: { [key: string]: any }) => data.LTSA_UPDATE_USER,
  },
  RELATED_GAZETTES: {
    label: 'Related gazettes',
    display: (data: { [key: string]: any }) => data.RELATED_GAZETTES,
  },
  RELATIVE_PATH: {
    label: 'Relative path',
    display: (data: { [key: string]: any }) => data.RELATIVE_PATH,
  },
};
