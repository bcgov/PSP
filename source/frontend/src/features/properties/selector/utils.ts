import { IMapProperty } from 'features/properties/selector/models';
import { compact } from 'lodash';
import { Api_PropertyFile } from 'models/api/PropertyFile';
import { formatApiAddress, pidFormatter } from 'utils';

import { Api_Geometry, Api_Property } from './../../../models/api/Property';

export enum NameSourceType {
  PID = 'PID',
  PIN = 'PIN',
  PLAN = 'Plan #',
  LOCATION = 'Location',
  NAME = 'Descriptive Name',
  NONE = 'n/a',
  ADDRESS = 'Address',
}

interface PropertyName {
  label: NameSourceType;
  value: string;
}

export const getPropertyName = (property: IMapProperty): PropertyName => {
  if (property.pid !== undefined && property.pid?.toString().length > 0 && property.pid !== '0') {
    return { label: NameSourceType.PID, value: pidFormatter(property.pid.toString()) };
  } else if (
    property.pin !== undefined &&
    property.pin?.toString()?.length > 0 &&
    property.pin !== '0'
  ) {
    return { label: NameSourceType.PIN, value: property.pin.toString() };
  } else if (property.planNumber !== undefined) {
    return { label: NameSourceType.PLAN, value: property.planNumber };
  } else if (property.latitude !== undefined && property.longitude !== undefined) {
    return {
      label: NameSourceType.LOCATION,
      value: compact([property.longitude?.toFixed(6), property.latitude?.toFixed(6)]).join(', '),
    };
  } else if (property.address !== undefined) {
    return {
      label: NameSourceType.ADDRESS,
      value: property.address,
    };
  }
  return { label: NameSourceType.NONE, value: '' };
};

export const getPrettyLatLng = (location?: Api_Geometry) =>
  compact([
    location?.coordinate?.x?.toFixed(6) ?? 0,
    location?.coordinate?.y?.toFixed(6) ?? 0,
  ]).join(', ');

export const getFilePropertyName = (
  fileProperty?: Api_PropertyFile,
  skipName: boolean = false,
): PropertyName => {
  if (fileProperty === undefined) {
    return { label: NameSourceType.NONE, value: '' };
  }

  if (
    fileProperty.propertyName !== undefined &&
    fileProperty.propertyName !== '' &&
    skipName === false
  ) {
    return { label: NameSourceType.NAME, value: fileProperty.propertyName };
  } else if (fileProperty.property !== undefined) {
    const property = fileProperty.property;
    return getApiPropertyName(property);
  }
  return { label: NameSourceType.NONE, value: '' };
};

export const getApiPropertyName = (property?: Api_Property): PropertyName => {
  if (property === undefined) {
    return { label: NameSourceType.NONE, value: '' };
  }

  let mapProperty: IMapProperty = {
    pin: property.pin?.toString(),
    pid: property.pid?.toString(),
    latitude: property.latitude,
    longitude: property.longitude,
    planNumber: property.planNumber,
    address: property.address !== undefined ? formatApiAddress(property.address) : undefined,
  };
  return getPropertyName(mapProperty);
};
