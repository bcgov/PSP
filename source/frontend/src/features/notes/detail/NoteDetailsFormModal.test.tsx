import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import { Claims } from 'constants/index';
import { mockLookups } from 'mocks/mockLookups';
import { mockNoteResponse } from 'mocks/mockNoteResponses';
import { Api_Note } from 'models/api/Note';
import { lookupCodesSlice } from 'store/slices/lookupCodes';
import { render, RenderOptions, userEvent, waitFor } from 'utils/test-utils';

import { INoteDetailsFormModalProps, NoteDetailsFormModal } from './NoteDetailsFormModal';

// mock auth library
jest.mock('@react-keycloak/web');

const mockAxios = new MockAdapter(axios);

const onEdit = jest.fn();
const onClose = jest.fn();

const BASIC_PROPS: INoteDetailsFormModalProps = {
  isOpened: true,
  loading: false,
  note: mockNoteResponse(1),
  onCloseClick: onClose,
  onEdit,
};

describe('NoteDetailsFormModal component', () => {
  // render component under test
  const setup = (
    props: INoteDetailsFormModalProps = { ...BASIC_PROPS },
    renderOptions: RenderOptions = {},
  ) => {
    const utils = render(<NoteDetailsFormModal {...props} />, {
      ...renderOptions,
      store: {
        [lookupCodesSlice.name]: { lookupCodes: mockLookups },
      },
      useMockAuthentication: true,
    });

    return {
      ...utils,
      getEditButton: () => utils.getByRole('button', { name: /edit/i }),
      getModalCloseButton: () => utils.getByTitle('ok-modal'),
    };
  };

  beforeEach(() => {
    mockAxios.onGet(new RegExp('users/info/*')).reply(200, {});
  });

  afterEach(() => {
    mockAxios.reset();
    jest.resetAllMocks();
  });

  it('renders as expected', () => {
    setup();
    expect(document.body).toMatchSnapshot();
  });

  it('renders a spinner while loading', () => {
    const { getByTestId } = setup({ ...BASIC_PROPS, loading: true });
    const spinner = getByTestId('filter-backdrop-loading');
    expect(spinner).toBeVisible();
  });

  it('renders the Note field', () => {
    const { getByTitle } = setup();
    const textarea = getByTitle(/Note/i);

    expect(textarea).toBeVisible();
    expect(textarea.tagName).toBe('TEXTAREA');
    expect(textarea).toHaveAttribute('readonly');
  });

  it('should execute callback when Close button is clicked', async () => {
    const { getModalCloseButton } = setup();
    userEvent.click(getModalCloseButton());

    expect(onClose).toBeCalled();
  });

  it(`should not display the Last Updated info for system-generated notes`, async () => {
    const systemNote: Api_Note = {
      ...mockNoteResponse(1),
      isSystemGenerated: true,
    };
    const { queryByText } = setup({ ...BASIC_PROPS, note: systemNote }, { claims: [] });
    const lastUpdated = queryByText(/last updated/i);
    expect(lastUpdated).toBeNull();
  });

  it(`should not render Edit button when user doesn't have edit permissions`, async () => {
    const { queryByRole } = setup({ ...BASIC_PROPS }, { claims: [] });
    const editNote = queryByRole('button', { name: /edit/i });
    expect(editNote).toBeNull();
  });

  it('should render Edit button when user has edit permissions', async () => {
    const { getEditButton } = setup({ ...BASIC_PROPS }, { claims: [Claims.NOTE_EDIT] });
    expect(getEditButton()).toBeVisible();
  });

  it('should execute callback when Edit button is clicked', async () => {
    const { getEditButton } = setup({ ...BASIC_PROPS }, { claims: [Claims.NOTE_EDIT] });
    await waitFor(() => userEvent.click(getEditButton()));

    expect(onEdit).toBeCalledWith(mockNoteResponse(1));
    expect(onClose).not.toBeCalled();
  });
});
