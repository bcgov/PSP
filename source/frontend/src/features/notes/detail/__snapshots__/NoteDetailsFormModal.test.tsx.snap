// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteDetailsFormModal component renders as expected 1`] = `
.c3.btn {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.4rem 1.2rem;
  min-height: 3rem;
  border: 0.2rem solid transparent;
  border-radius: 0.4rem;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.8rem;
  font-family: 'BCSans','Noto Sans',Verdana,Arial,sans-serif;
  font-weight: 700;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
  cursor: pointer;
}

.c3.btn:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  opacity: 0.8;
}

.c3.btn:focus {
  outline-width: 0.4rem;
  outline-style: solid;
  outline-offset: 1px;
  box-shadow: none;
}

.c3.btn.btn-primary {
  border: none;
}

.c3.btn.btn-secondary {
  background: none;
}

.c3.btn.btn-info {
  border: none;
  background: none;
  padding-left: 0.6rem;
  padding-right: 0.6rem;
}

.c3.btn.btn-info:hover,
.c3.btn.btn-info:active,
.c3.btn.btn-info:focus {
  background: none;
}

.c3.btn.btn-light {
  border: none;
}

.c3.btn.btn-dark {
  border: none;
}

.c3.btn.btn-link {
  font-size: 1.6rem;
  font-weight: 400;
  background: none;
  border: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  min-height: 2.5rem;
  line-height: 3rem;
  -webkit-box-pack: left;
  -webkit-justify-content: left;
  -ms-flex-pack: left;
  justify-content: left;
  -webkit-letter-spacing: unset;
  -moz-letter-spacing: unset;
  -ms-letter-spacing: unset;
  letter-spacing: unset;
  text-align: left;
  padding: 0;
}

.c3.btn.btn-link:hover,
.c3.btn.btn-link:active,
.c3.btn.btn-link:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  border: none;
  background: none;
  box-shadow: none;
  outline: none;
}

.c3.btn.btn-link:disabled,
.c3.btn.btn-link.disabled {
  background: none;
  pointer-events: none;
}

.c3.btn:disabled,
.c3.btn:disabled:hover {
  box-shadow: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  pointer-events: none;
  cursor: not-allowed;
  opacity: 0.65;
}

.c3.Button .Button__icon {
  margin-right: 1.6rem;
}

.c3.Button--icon-only:focus {
  outline: none;
}

.c3.Button--icon-only .Button__icon {
  margin-right: 0;
}

.c0 .modal-header .close {
  color: white;
  opacity: 1;
  font-size: 3rem;
  font-weight: 10;
  text-shadow: none;
  font-family: 'Helvetica Narrow';
}

.c0 .modal-header {
  height: 4.8rem;
  padding: 0 1.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .modal-header .modal-title {
  font-family: BcSans-Bold;
  font-size: 2.2rem;
  height: 2.75rem;
  height: auto;
}

.c0 .modal-header .header-icon {
  margin-right: 8px;
  display: inline-block;
}

.c0 .modal-header .modal-close-btn {
  cursor: pointer;
}

.c0 .modal-body {
  padding: 2.4rem 1.8rem;
  font-size: 1.8rem;
}

.c0 .modal-footer {
  border-top: none;
}

.c0 .modal-footer hr {
  width: 100%;
}

.c0 .modal-footer .button-wrap {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-top: 2.4rem;
  margin-bottom: 2.4rem;
}

.c0 .modal-footer .button-wrap button {
  margin-right: 2.4rem;
  min-width: 9.5rem;
  height: 3.9rem;
}

.c0 .close {
  color: black;
}

.c0 .modal-xl {
  max-width: 100rem;
}

.c0 .modal-l {
  max-width: 75rem;
}

.c0 .modal-m {
  max-width: 50rem;
}

.c0 .modal-s {
  max-width: 40rem;
}

.c0.info-variant .modal-close-btn {
  cursor: pointer;
}

.c0.error-variant .modal-close-btn {
  cursor: pointer;
}

.c0.warning-variant .modal-close-btn {
  cursor: pointer;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: end;
  -ms-flex-pack: end;
  justify-content: end;
}

.c1 {
  min-width: 70rem;
}

.c1 .modal-body {
  padding-left: 2rem;
  padding-right: 2rem;
}

.c1 .modal-footer {
  padding-left: 2rem;
  padding-right: 2rem;
}

.c1 .form-group label {
  font-family: BcSans-Bold;
  line-height: 2rem;
}

<body
  class="modal-open"
  style="padding-right: 0px;"
>
  <div
    aria-hidden="true"
  >
    <div
      class="Toastify"
    />
    <div />
    <div />
  </div>
  <div
    class="fade modal-backdrop show"
  />
  <div
    aria-modal="true"
    class="fade c0 c1 info-variant modal show"
    role="dialog"
    style="display: block;"
    tabindex="-1"
  >
    <div
      class="modal-dialog c1 info-variant"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <div
            class="modal-title h4"
          >
            Notes
          </div>
          <button
            class="close"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              Ã—
            </span>
            <span
              class="sr-only"
            >
              Close
            </span>
          </button>
        </div>
        <div
          class="modal-body"
          style="white-space: pre-line;"
        >
          <div
            class="container"
          >
            <div
              class="no-gutters row"
            >
              <div
                class="mr-2 col-md-2"
              >
                Created:
              </div>
              <div
                class="col"
              >
                <span>
                  <strong>
                    Jul 20, 2022
                  </strong>
                   by
                   
                  <span
                    class="tooltip-icon"
                    data-testid="tooltip-icon-userNameTooltip"
                    id="userNameTooltip"
                  >
                    <strong>
                      admin
                    </strong>
                  </span>
                </span>
              </div>
            </div>
            <div
              class="no-gutters row"
            >
              <div
                class="mr-2 col-md-2"
              >
                Last updated:
              </div>
              <div
                class="col"
              >
                <span>
                  <strong>
                    Jul 20, 2022
                  </strong>
                   by
                   
                  <span
                    class="tooltip-icon"
                    data-testid="tooltip-icon-userNameTooltip"
                    id="userNameTooltip"
                  >
                    <strong>
                      admin
                    </strong>
                  </span>
                </span>
              </div>
              <div
                class="c2 col"
              />
            </div>
            <div
              class="no-gutters row"
            >
              <div
                class="col"
              >
                <textarea
                  class="form-control"
                  readonly=""
                  rows="15"
                  title="Note"
                >
                  Test Note
                </textarea>
              </div>
            </div>
          </div>
        </div>
        <div
          class="modal-footer"
        >
          <hr />
          <div
            class="button-wrap"
          >
            <button
              class="c3 Button btn btn-primary"
              data-testid="ok-modal-button"
              title="ok-modal"
              type="button"
            >
              <div
                class="Button__value"
              >
                Close
              </div>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
